### PR TITLE
Fix Quality Flaws: simplify regexes

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/AbstractPrintfChecker.java [hunk 1]
```diff
@@ -57,7 +57,12 @@ public abstract class AbstractPrintfChecker extends AbstractMethodDetection {
   protected static final String JAVA_LANG_THROWABLE = "java.lang.Throwable";
   protected static final String ORG_APACHE_LOGGING_LOG4J_LOGGER = "org.apache.logging.log4j.Logger";
 
-  private static final Pattern PRINTF_PARAM_PATTERN = Pattern.compile("%(\\d+\\$)?([-#+ 0,(\\<]*)?(\\d+)?(\\.\\d+)?([tT])?([a-zA-Z%])");
+  private static final String ARGUMENT_INDEX = "(\\d+\\$)?";
+  private static final String FLAGS = "([-#+ 0,(\\<]+)?";
+  private static final String WIDTH = "(\\d+)?";
+  private static final String PRECISION = "(\\.\\d+)?";
+  private static final String CONVERSION = "([tT])?([a-zA-Z%])";
+  private static final Pattern PRINTF_PARAM_PATTERN = Pattern.compile("%" + ARGUMENT_INDEX + FLAGS + WIDTH + PRECISION + CONVERSION);
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/AbstractPrintfChecker.java:65]: reduce cognitive complexity

File: java-checks/src/main/java/org/sonar/java/checks/StringMethodsWithLocaleCheck.java [hunk 2]
```diff
@@ -71,9 +71,9 @@ private static class FormatterVisitor extends BaseTreeVisitor {
 
     // source: https://docs.oracle.com/javase/8/docs/api/java/util/Formatter.html#syntax
     private static final String ARGUMENT_INDEX = "(\\d+\\$)?";
-    private static final String FLAGS = "([-,#+ 0(]*)?";
+    private static final String FLAGS = "([-,#+ 0(]+)?";
     private static final String FLAGS_WITH_FORCED_COMMA = "([-#+ 0(]*,[-#+ 0(]*)";
-    private static final String WIDTH = "([0-9]*\\.?[0-9]*)?";
+    private static final String WIDTH = "(\\d*\\.?\\d*)";
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/StringMethodsWithLocaleCheck.java:76]: replace group

File: java-checks/src/main/java/org/sonar/java/checks/StringMethodsWithLocaleCheck.java [hunk 3]
```diff
@@ -71,9 +71,9 @@ private static class FormatterVisitor extends BaseTreeVisitor {
 
     // source: https://docs.oracle.com/javase/8/docs/api/java/util/Formatter.html#syntax
     private static final String ARGUMENT_INDEX = "(\\d+\\$)?";
-    private static final String FLAGS = "([-,#+ 0(]*)?";
+    private static final String FLAGS = "([-,#+ 0(]+)?";
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/StringMethodsWithLocaleCheck.java:74]: make sure it does not match empty string (`?` is there for that)

File: java-checks/src/main/java/org/sonar/java/checks/security/FilePermissionsCheck.java [hunk 4]
```diff
@@ -55,8 +55,13 @@ public class FilePermissionsCheck extends IssuableSubscriptionVisitor {
...
   // 'other' group not being 0
   private static final Pattern CHMOD_OCTAL_PATTERN = Pattern.compile("(^|\\s)[0-7]{2,3}[1-7](\\s|$)");
+
+  private static final String WHO = "([ug]*+[ao][ugao]*+)?";
+  // ignoring '-' as it should be safe to explicitly REMOVE permission
+  private static final String WHAT = "[+=]";
+  private static final String WHICH = "[sStT]*+[rwxX][rwxXsStT]*+";
   // simplification of all the possible combinations of adding perms to 'other'
-  private static final Pattern SIMPLIFIED_CHMOD_OTHER_PATTERN = Pattern.compile("(^|\\s|,)([ug]*+[ao][ugao]*+)?[+=][sStT]*+[rwxX][rwxXsStT]*+(\\s|,|$)");
+  private static final Pattern SIMPLIFIED_CHMOD_OTHER_PATTERN = Pattern.compile("(^|\\s|,)" + WHO + WHAT + WHICH + "(\\s|,|$)");
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/security/FilePermissionsCheck.java:64]: reduce cognitive complexity

File: java-frontend/src/main/java/org/sonar/java/model/LiteralUtils.java [hunk 5]
```diff
@@ -171,7 +171,7 @@ private static String stripIndent(int indent, String s) {
   }
 
   private static String removeTrailingSpaces(String s) {
-    return s.replaceAll("[\t \f]*$", "");
+    return s.replaceAll("\\s+$", "");
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/model/LiteralUtils.java:174]: remove catastrophic backtracking risk ([java:S5852](https://next.sonarqube.com/sonarqube/coding_rules?open=java%3AS5852&rule_key=java%3AS5852))

Comment (alban-auzeill) [reply java-frontend/src/main/java/org/sonar/java/model/LiteralUtils.java:174]: The fix should be `[\t \f]++$`

**Review by alban-auzeill — APPROVED** [review]

**Pull request merged**